src/include/migraphx/module.hpp

```diff
@@ -205,6 +209,28 @@ struct MIGRAPHX_EXPORT module
 
     void finalize(std::vector<context>& contexts);
 
+    /// Create a mapping from the input instruction to the corresponding
+    /// parameter instruction. Use the `reverse` flag to reverse the lookup
+    /// to be from parameter instruction to input instread.
+    std::unordered_map<instruction_ref, instruction_ref>
+    get_ins_param_map(const std::vector<instruction_ref>& inputs, bool reverse = false) const;
+
+    using with_inputs = module_with_inputs;
+
+    /// This will split the module into two parts at the instruction splits.
+    /// Each split instruction becomes an input parameter in the second
+    /// module. As such the inputs instructions to the second module will use
+    /// the split instructions as input placeholders that can be replaced
+    /// later.
```

Comment:
Thanks for this. It helped understanding what it is trying to do instead of guessing what it is doing. 

We should add unit-tests for this

Reply:
Yea I will add unit tests for this.